core/lib/state/src/lib.rs

```diff
@@ -87,3 +90,9 @@ pub trait WriteStorage: ReadStorage {
 
 /// Smart pointer to [`WriteStorage`].
 pub type StoragePtr<S> = Rc<RefCell<S>>;
+
+/// Functionality to override the storage state.
+pub trait OverrideStorage {
+    /// Apply state override to the storage.
+    fn apply_state_override(&mut self, overrides: &StateOverride);
+}
```

Comment:
we have the only implementation and no generics over this trait, why it's not just part of the struct impl?

Reply:
I think it's ok to implement it for storage view. 

and call override storage from there